internal/app/machined/pkg/runtime/v1alpha1/platform/metal/metal.go

```diff
@@ -45,6 +45,10 @@ func (m *Metal) Configuration(ctx context.Context) ([]byte, error) {
 		return nil, errors.ErrNoConfigSource
 	}
 
+	if *option == constants.ConfigNone {
+		return nil, errors.ErrNoConfigSource
+	}
```

Comment:
I wonder if this will break subsequent boots. What happens after applying a config, and this arg is not updated?

Reply:
we don't drop into this function unless configuration is missing

so if we got here, we need to fetch config, as config is missing. we can't fetch it from`none` obviously

Reply:
ah... 👍🏻